packages/app-builder/src/models/data-model.ts

```diff
@@ -644,3 +645,34 @@ export function adaptCreateAnnotationDto(model: CreateAnnotationBody): CreateAnn
     )
     .exhaustive();
 }
+
+export type IngestedDataField = {
+  path: string[];
+  name: string;
+};
+
+export type ExportedFields = {
+  triggerObjectFields: string[];
+  ingestedDataFields: IngestedDataField[];
+};
+
+export function adaptExportedFieldsDto(dto: ExportedFieldsDto): ExportedFields {
```

Comment:
I'm kinda nitpicking on the naming of the adapters, but it's not consistent with other function (same for the transform below).
Usually when doing `EntityDto -> Entity` the function is named `adaptEntity` and when doing it the other way (`Entity -> EntityDto`) it's named `adaptEntityDto`.

Reply:
oh sorry I thought we had an agreement on introducing adapt/transform function prefixes depending on the DTO/entity way we move. My bad